packages/flutter/lib/src/widgets/app.dart

```diff
@@ -149,6 +185,45 @@ class WidgetsApp extends StatefulWidget {
   /// for this application's [Localizations] widget.
   final Iterable<LocalizationsDelegate<dynamic>> localizationsDelegates;
 
+  /// This callback is responsible for choosing the app's locale
+  /// when the app is started, and when the user changes the
+  /// device's locale.
+  ///
+  /// The returned value becomes the locale of this app's [Localizations]
+  /// widget. The `oldLocale` parameter is null when the app starts. The
+  /// `newLocale` is the device's locale when the app started, or the
+  /// device locale the user selected after the app was started.
+  /// The `supportedLocales` parameter is just the value [supportedLocales].
+  ///
+  /// This callback must not be null. Its default value is
+  /// [defaultLocaleChangedHandler].
+  ///
+  /// See also:
+  ///
+  ///  * [MaterialApp.onLocaleChanged], which sets the callback of the
+  ///    [WidgetsApp] it creates.
+  final LocaleChangedCallback onLocaleChanged;
```

Comment:
maybe call this localeResolutionCallback so it's more specifically about locale resolution rather than about the change in system locale?

Reply:
I ended up calling it resolveLocaleCallback, although localeResolutionCallback might have been better.

Reply:
Generally I would recommend the form "nounCallback" rather than "verbCallback", to distinguish it from a method (resolveLocaleCallback sounds like a method that resolves the locale callback).

Reply:
Since there are other things to change, I vote for changing the name here too while you're at it.